### PR TITLE
Limit max number of layers

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Add restriction to create more than 100 layers.

Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

I took snapshot with 3 as max limit to avoid creating 20 layers

Screen Shot 2023-01-18 at 5 41 41 PM

### Issues Resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by junqiu-lei** [comment]
Need we add a warning message when user reached the max number?

File: common/index.ts [hunk 1]
```diff
@@ -23,25 +23,26 @@ export {
...
+export const MAP_DEFAULT_MIN_ZOOM = 0;
+export const MAP_LAYER_DEFAULT_BORDER_THICKNESS = 1;
+export const MAP_LAYER_DEFAULT_MAX_OPACITY = 100;
+export const MAP_LAYER_DEFAULT_MIN_OPACITY = 0;
 export const MAP_LAYER_DEFAULT_NAME = 'Default map';
-export const NEW_MAP_LAYER_DEFAULT_PREFIX = 'New layer';
-export const MIN_LONGITUDE = -180;
+export const MAP_LAYER_DEFAULT_OPACITY_STEP = 1;
+export const MAP_REFERENCE_LAYER_DEFAULT_OPACITY = 100;
+export const MAX_LAYER_LIMIT = 100; // Move this setting to Map configuration later
```
Comment (navneet1v) [inline common/index.ts:41]: This setting of 100 is too big for the UI. I would recommend making a default setting as 20(no great reason to support this except that it is small and will make sure that UI is responsive.)

Also, given that whatever value we choose we may endup debating what is the right default value. I would say let's get this value from config file so that customer have a way to update this value if required.

File: common/index.ts [hunk 2]
```diff
@@ -23,25 +23,26 @@ export {
...
+export const MAX_LAYER_NAME_LIMIT = 35;
```
Comment (navneet1v) [inline common/index.ts:42]: Please add comments why we are using a default value. Your comment can say that this is default value for other components of dashboard will also be fine.

File: public/components/add_layer_panel/add_layer_panel.tsx [hunk 3]
```diff
@@ -108,7 +110,7 @@ export const AddLayerPanel = ({
           fill
           onClick={showModal}
           aria-label="Add layer"
-          isDisabled={IsLayerConfigVisible}
+          isDisabled={IsLayerConfigVisible || layerCount >= MAX_LAYER_LIMIT}
```
Comment (navneet1v) [inline public/components/add_layer_panel/add_layer_panel.tsx:113]: when the add layer button is disabled, I am not able to see the tool-tip which provides info around why this button is disabled. If it is not added, please add the tool-tip, otherwise please refer the code here for the tool tip.

File: common/index.ts (hunk 1 above)
Comment (VijayanB) [reply common/index.ts:41]: Agree. In future this value will be part of Map settings for users to configure.

File: public/components/add_layer_panel/add_layer_panel.tsx (hunk 3 above)
Comment (VijayanB) [reply public/components/add_layer_panel/add_layer_panel.tsx]: Added. I was waiting for text review .

File: common/index.ts (hunk 2 above)
Comment (VijayanB) [reply common/index.ts:43]: Do you want comment for layer name or layer limit?

Comment (navneet1v) [reply common/index.ts:43]: layer name size limit

File: common/index.ts (hunk 1 above)
Comment (navneet1v) [reply common/index.ts]: Why we are not making this setting as of now and putting this as a part of future work?

Comment (VijayanB) [reply common/index.ts]: Map settings needs UX.

Comment (navneet1v) [reply common/index.ts]: wait there is a gap in the understanding. Let me rephrase what I was saying: I was saying we can take this settings from opensearch_dashboards.yml file.

Comment (VijayanB) [reply common/index.ts]: I don't think we need this to be part of osd yaml file. I am thinking about adding as ui settings similar to other dashboard objects. This will provide users to change this inside Dashboards instead of updating in yaml file, which requires OSD restart i believe

Comment (navneet1v) [reply common/index.ts]: I agree that it will require an OSD bootstrap and making it as a dynamic settings that can be updated using UI will be great. Please point me to the github issue which we are using to track you proposed approach.

**Review by navneet1v — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

File: common/index.ts (hunk 1 above)
Comment (VijayanB) [reply common/index.ts]: https://github.com/opensearch-project/dashboards-maps/issues/217
